core/factorization/cholesky.cpp

```diff
@@ -36,6 +39,27 @@ GKO_REGISTER_OPERATION(factorize, cholesky::factorize);
 }  // namespace
 
 
+template <typename ValueType, typename IndexType>
+typename Cholesky<ValueType, IndexType>::parameters_type
+Cholesky<ValueType, IndexType>::parse(
+    const config::pnode& config, const config::registry& context,
+    const config::type_descriptor& td_for_child)
+{
+    using sparsity_pattern_type =
+        Cholesky<ValueType, IndexType>::sparsity_pattern_type;
+    auto params = Cholesky<ValueType, IndexType>::build();
+    if (auto& obj = config.get("symbolic_factorization")) {
```

Comment:
we have some options that require their values to be keys in the registry. Maybe we need to specify this somewhere. It should only concern parameters which are stored as linops in the factory, so maybe we can add somthing about this to the guidelines.